### PR TITLE
Open email app in compose view with content

### PR DESCRIPTION
**Pull request by diskwriter**
Building on the work started by @azorychta. All of the launch schemes were taken from their PR.

Built in collaboration with @krispypen. 
Supported on android and iOS for the `to`, `cc`, `bcc`, `subject` and `body` fields.
 
Should close https://github.com/HomeXLabs/open-mail-app-flutter/issues/6

### CODE REVIEW

**Comment by MisterJimson** [comment]
Wow! Thank you. We will take a look very soon.

File: android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt [hunk 1]
```diff
@@ -65,6 +66,20 @@ class OpenMailAppPlugin : FlutterPlugin, MethodCallHandler {
...
+            } else {
+                result.success(false)
+            }
+        } else if (call.method == "composeNewEmailInSpecificMailApp") {
+            val opened = composeNewEmailInSpecificEmailAppIntent(call.argument("name") ?: "", call.argument("emailContent") ?: "")
+            if (opened) {
+                result.success(true);
+            } else {
+                result.success(false);
+            }
```
Comment (dw2kim) [inline android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt:82]:
```suggestion
            result.success(opened)
```

File: android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt [hunk 2]
```diff
@@ -14,6 +14,7 @@ import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 import io.flutter.plugin.common.PluginRegistry.Registrar
+import java.net.URLDecoder
```
Comment (dw2kim) [inline android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt:17]: nit: Unused import

File: android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt (hunk 1 above)
Comment (dw2kim) [reply android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt:82]: maybe for all the if statements here

File: android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt [hunk 3]
```diff
@@ -115,6 +130,58 @@ class OpenMailAppPlugin : FlutterPlugin, MethodCallHandler {
...
+                                    data = Uri.parse("mailto:")
+                                    type = "text/plain"
+                                    setClassName(activityHandlingEmail.activityInfo.packageName, activityHandlingEmail.activityInfo.name)
+                                    putExtra(Intent.EXTRA_EMAIL, emailContent.to.toTypedArray())
+                                    putExtra(Intent.EXTRA_CC, emailContent.cc.toTypedArray())
+                                    putExtra(Intent.EXTRA_BCC, emailContent.bcc.toTypedArray())
+                                    putExtra(Intent.EXTRA_SUBJECT, emailContent.subject)
+                                    putExtra(Intent.EXTRA_TEXT, emailContent.body)
+                                },
+                            packageName,
```
Comment (dw2kim) [inline android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt:168]: nit: According to `ktlint`, no trailing space
```suggestion
                            packageName,
```

File: android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt [hunk 4]
```diff
@@ -153,4 +248,13 @@ class OpenMailAppPlugin : FlutterPlugin, MethodCallHandler {
...
+)
+
+data class EmailContent (
+
+        @SerializedName("to") val to: List<String>,
+        @SerializedName("cc") val cc: List<String>,
+        @SerializedName("bcc") val bcc: List<String>,
+        @SerializedName("subject") val subject: String,
+        @SerializedName("body") val body: String
 )
```
Comment (dw2kim) [inline android/src/main/kotlin/com/homex/open_mail_app/OpenMailAppPlugin.kt:260]: nit: According to ktlint,  File must end with a newline (\n)
```suggestion
)

```

**Review by dw2kim — COMMENTED** [review]
Looks amazing after running some tests. Left some comments. 

➕  it would great if we can update the README for the Usage part.

**Comment by MisterJimson** [comment]
@diskwriter thanks again for the PR! Let us know if you are happy to work on the feedback or if you would prefer we do it.

**Comment by diskwriter** [comment]
@dw2kim @MisterJimson 

Thanks a lot for the review! Happy to work on the feedback; I'll pick it up somewhere tomorrow.

File: lib/open_mail_app.dart [hunk 5]
```diff
@@ -151,7 +298,15 @@ class MailAppPickerDialog extends StatelessWidget {
...
             onPressed: () {
-              OpenMailApp.openSpecificMailApp(app);
+              if (emailContent == null) {
+                OpenMailApp.openSpecificMailApp(app);
+              } else {
+                OpenMailApp.composeNewEmailInSpecificMailApp(
+                  mailApp: app,
+                  emailContent: emailContent!,
+                );
+              }
```
[the remark this answers is not in the document]
Comment (diskwriter) [reply lib/open_mail_app.dart]: Sure, I prefer not to do this on properties which are located on `this` but I've changed it to follow convention of this project.

**Comment by diskwriter** [comment]
@dw2kim Resolved your comments! Could you take another look? Thanks!

cc/ @MisterJimson

**Review by dw2kim — APPROVED** [review]
LGTM!

**Pull request merged**